python/test/unit/language/test_core.py

```diff
@@ -1928,7 +1930,8 @@ def kernel(X, Z, BLOCK_M: tl.constexpr, BLOCK_N: tl.constexpr, BLOCK_K: tl.const
     BLOCK_K = 1 if len(shape) == 2 else shape[2]
     IS_3D = bool(len(shape) == 3)
     if axis is not None and axis >= len(shape):
-        with pytest.raises(triton.CompilationError):
+        error_class = ValueError if is_interpreter() else triton.CompilationError
```

Comment:
@jlebar Sorry again for the hack. I'll make sure to clean this up after syncing with Phil next week.